XmlFormatter/src/Hotfolder/HotfolderManager.cs

```diff
@@ -146,15 +152,43 @@ private void FormatterToUse_StatusChanged(object sender, EventMessages.BaseEvent
         /// Get the hotfolder by the watched folder
         /// </summary>
         /// <param name="watchedFolder">The watched folder to get the hotfolder for</param>
+        /// <param name="filter">The filter used by the watcher</param>
         /// <returns>The hotfolder</returns>
-        private IHotfolder GetHotfolderByWatchedFolder(string watchedFolder)
+        /// //
+        private IHotfolder GetHotfolderByWatchedFolder(string watchedFolder, string filter)
         {
             return hotfolders.Keys.ToList().Find((currentHotfolder) =>
             {
-                return currentHotfolder.WatchedFolder == watchedFolder;
+                return currentHotfolder.WatchedFolder == watchedFolder && currentHotfolder.Filter == filter;
             });
         }
 
+        /// <summary>
+        /// Is there already an identical watcher registerec
+        /// </summary>
+        /// <param name="hotfolder">The new hotfolder config to check</param>
+        /// <returns>True if there is already an identical active configuration</returns>
+        private bool IsSameWatcherRegisterd(IHotfolder hotfolder)
+        {
+            return hotfolders.Keys.ToList().Find(currentHotfolder =>
+            {
+                bool isIdentical = false;
+                if (hotfolder.FormatterToUse == null)
+                {
+                    return true;
+                }
+                if (currentHotfolder.FormatterToUse.GetType() == hotfolder.FormatterToUse.GetType())
+                {
+                    if (currentHotfolder.Mode == hotfolder.Mode && currentHotfolder.WatchedFolder == hotfolder.WatchedFolder)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Merge this if statement with the enclosing one.](https://app.codacy.com/manual/simonaberle/XmlFormatter/pullRequest?prid=5553221)